computer_vision/harriscorner.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+import cv2
+
+"""
+Harris Corner Detector
+https://en.wikipedia.org/wiki/Harris_Corner_Detector
+"""
+
+
+class Harris_Corner:
+    def __init__(self, k, window_size):
+
+        if k in [0.04, 0.06]:
+            self.k = k
+            self.window_size = window_size
+        else:
+            print("invalid k value")
+            exit()
+
+    def __str__(self):
+        return "Harris Corner  detection with k : {}".format(self.k)
+
+    def detect(self, img_path):
+        """
+        Returns the image with corners identified
+        img_path  : path of the image
+        output : list of the corner positions, image
+        """
+        img = cv2.imread(img_path, 0)
+        h, w = img.shape
+        corner_list = []
+        color_img = img.copy()
+        color_img = cv2.cvtColor(color_img, cv2.COLOR_GRAY2RGB)
```

Comment:
Is this converting the image from grayscale to RGB or is it turning RGB to grayscale? The wikipedia page says to turn it to grayscale.

Reply:
Actually that's for the output image.